src/core/Decorations.ts

```diff
@@ -261,25 +258,24 @@ export default class Decorations {
   ) {
     const iconPath = join(this.extensionPath, "images", "hats", `${shape}.svg`);
     const rawSvg = readFileSync(iconPath, "utf8");
+    const { characterWidth, characterHeight } = fontMeasurements;
 
     const { originalViewBoxHeight, originalViewBoxWidth } =
       this.getViewBoxDimensions(rawSvg);
 
-    const hatWidthPx =
-      hatWidthToCharacterWidthRatio * fontMeasurements.characterWidth;
+    const hatWidthPx = hatWidthToCharacterWidthRatio * characterWidth;
     const hatHeightPx =
       (originalViewBoxHeight / originalViewBoxWidth) * hatWidthPx;
 
-    const svgWidthPx = Math.ceil(fontMeasurements.characterWidth);
-    const svgHeightPx =
-      fontMeasurements.characterHeight + hatHeightPx + hatVerticalOffset;
+    const svgWidthPx = Math.ceil(characterWidth);
+    const svgHeightPx = characterHeight + hatHeightPx + hatVerticalOffset;
 
     const newViewBoxWidth =
-      ((originalViewBoxWidth / hatWidthToCharacterWidthRatio) *
-        fontMeasurements.characterWidth) /
-      svgWidthPx;
+      ((originalViewBoxWidth / hatWidthToCharacterWidthRatio) * svgWidthPx) /
+      characterWidth;
     const newViewBoxHeight = (newViewBoxWidth * svgHeightPx) / svgWidthPx;
-    const newViewBoxX = -(newViewBoxWidth - originalViewBoxWidth) / 2;
+    const newViewBoxX =
+      (-(characterWidth - hatWidthPx) * (newViewBoxWidth / svgWidthPx)) / 2;
```

Comment:
Since we're left-aligned, we basically need to shift right by `(characterWidth - hatWidthPx) / 2`, translated to svg view box space (`newViewBoxWidth / svgWidthPx`)